lib/mayaUsd/ufe/private/Utils.cpp

```diff
@@ -193,6 +194,18 @@ void applyCommandRestriction(const UsdPrim& prim, const std::string& commandName
     }
 }
 
+bool applyCommandRestrictionNoThrow(const UsdPrim& prim, const std::string& commandName)
```

Comment:
The applyCommandRestriction method is rather surprising. It uses an exception mechanism to apply restrictions. It's not a pattern we would like to repeat but since this code was here before, let's consider it technical debt. Next time  we are here, let's swap the order and make the throwing version a wrapper on top of a method that explicitly returns a state.